src/startupscriptgenerator/python/scriptgenerator.go

```diff
@@ -93,75 +94,84 @@ func logReadDirError(logger *common.Logger, path string, err error) {
-			scriptBuilder.WriteString("    echo \"Extracting...\"\n")
-			scriptBuilder.WriteString("    $extractionCommand\n")
-			scriptBuilder.WriteString("	   virtualEnvDir=\"/$virtualEnvName\"\n")
-			scriptBuilder.WriteString("elif [ ! -z \"$virtualEnvName\" ]; then\n")
-			scriptBuilder.WriteString("    virtualEnvDir=\"" + virtualEnvVarPath + "\"\n")
-			scriptBuilder.WriteString("fi\n")
-			virtualEnvCommand := getVirtualEnvironmentCommand()
-			scriptBuilder.WriteString(virtualEnvCommand)
-		}
-	} else {
-		// For backward compatibility, we check for the python packages folder.
-		packageDir := filepath.Join(gen.SourcePath, gen.PackageDirectory)
-		if common.PathExists(packageDir) {
-			scriptBuilder.WriteString("echo \"Using package directory '" + packageDir + "'\"\n" +
-				"SITE_PACKAGE_PYTHON_VERSION=$(python -c \"import sys; print(str(sys.version_info.major) + '.' + str(sys.version_info.minor))\")\n" +
-				"SITE_PACKAGES_PATH=$HOME\"/.local/lib/python\"$SITE_PACKAGE_PYTHON_VERSION\"/site-packages\"\n" +
-				"mkdir -p $SITE_PACKAGES_PATH\n" +
-				"echo \"" + packageDir + "\" > $SITE_PACKAGES_PATH\"/oryx.pth\"\n" +
-				"PATH=\"" + packageDir + "/bin:$PATH\"\n")
-		} else if gen.VirtualEnvironmentName != "" {
-			virtualEnvDir := filepath.Join(gen.SourcePath, gen.VirtualEnvironmentName)
-			if common.PathExists(virtualEnvDir) {
-				// We add the virtual env site-packages to PYTHONPATH instead of activating it to be backwards compatible with existing
-				// app service implementation. If we activate the virtual env directly things don't work since it has hardcoded references to
-				// python libraries including the absolute path. Since Python is installed in different paths in build and runtime images,
-				// the libraries are not found.
-				scriptBuilder.WriteString("virtualEnvDir=\"" + filepath.Join(gen.SourcePath, gen.VirtualEnvironmentName) + "\"\n")
-				scriptBuilder.WriteString("virtualEnvName=\"" + gen.VirtualEnvironmentName + "\"\n")
-				virtualEnvCommand := getVirtualEnvironmentCommand()
-				scriptBuilder.WriteString(virtualEnvCommand)
-
-			} else {
-				// We just warn the user and don't error out, since we still can run the default website.
-				scriptBuilder.WriteString("  echo \"WARNING: Could not find packages folder or virtual environment.\"\n")
-			}
-		}
-	}
-	return scriptBuilder.String()
-}
-
-func getVirtualEnvironmentCommand() string {
-	scriptBuilder := strings.Builder{}
-	scriptBuilder.WriteString("PYTHON_VERSION=$(python -c \"import sys; print(str(sys.version_info.major) + '.' + str(sys.version_info.minor))\")\n")
-	scriptBuilder.WriteString("echo \"Using packages from virtual environment '$virtualEnvName' located at '$virtualEnvDir'.\"\n")
-	virtualEnvFolder := "\"$virtualEnvDir/lib/python$PYTHON_VERSION/site-packages\""
-	scriptBuilder.WriteString("export PYTHONPATH=$PYTHONPATH:" + virtualEnvFolder + "\n")
+	scriptBuilder.WriteString("sourcePath=\"" + gen.SourcePath + "\"\n")
```

Comment:
why are you changing the entire block to bash? The final script should be as simple as possible.

Reply:
I changed it to make it more readable as before this we shift between golang and bash which I felt a bit hard to follow. the code structure that I have here is similar to what I did earlier for dotnetcore. The declarations of these variable can be viewed as inputs to a bash script which follows these declarations.
I also ran nightly build against this PR to make sure I did not break any tests.